apps/dashboard/src/components/workflow-editor/steps/in-app/in-app-tabs.tsx

```diff
@@ -39,18 +39,22 @@ export const InAppTabs = ({ workflow, step }: { workflow: WorkflowResponseDto; s
 
   const { dataSchema, uiSchema, values } = step.controls;
   const schema = useMemo(() => buildDynamicZodSchema(dataSchema ?? {}), [dataSchema]);
-  const newFormValues = useMemo(() => merge(buildDefaultValues(uiSchema ?? {}), values), [uiSchema, values]);
+  const newFormValues = useMemo(
+    () => merge(buildDefaultValues(uiSchema ?? {}), buildDefaultValuesOfDataSchema(dataSchema ?? {}), values),
+    [uiSchema, values, dataSchema]
+  );
 
   const form = useForm({
     mode: 'onSubmit',
     resolver: zodResolver(schema),
     values: newFormValues,
+    defaultValues: newFormValues,
```

Comment:
Any reason why we need the default values?

Reply:
This is leftover from the rebase, updated it now to reflect the changes on next branch